src/protocol.rs

```diff
@@ -48,6 +48,28 @@ pub enum ProtocolType {
 /// takes as input a [`Message`] and outputs a tuple containing the
 /// participant's output alongside a list of messages to process.
 ///
+/// # 🔒 Message handling
+/// The calling application is responsible for all message sending and routing
+/// for its [`Participant`]. This includes:
+/// 1. Initializing a session by calling [`Participant::initialize_message()`].
+///     The message returned from that function must be passed back to the
+/// [`Participant`] in order to begin the protocol execution.
```

Comment:
This shouldn't be fixed in this PR, but this is super funky. I think we should just have a constructor that creates a pre-initialized (Internally calls `initialize_message` and passes the message to the participant) `Participant` and returns it.

Reply:
I'll move this to a different issue so we don't block this PR :)